### ISSUE TITLE
Extra  chars in `/etc/hosts`

### ISSUE THREAD

**Issue by elsaco** [microsoft/WSL#9642]
### Version

Microsoft Windows [Version 10.0.19045.2604]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

AlmaLinux 9.1

### Repro Steps

Check system logs for `bad address at /etc/hosts` kind of message (i.e. journalctl --grep "bad address")

### Expected Behavior

No extra characters added to /etc/hosts

### Actual Behavior

`<feff>` added to /etc/hosts. Top pane show Windows' hosts file content. The bottom is WSL, Use `view` to display as `cat` or `bat` won't show it. Notice line 6:

![wsl_hosts](https://user-images.githubusercontent.com/3933920/219244086-4a66c490-586f-4a3c-86cd-8418aada1b18.png)

**Comment by pmartincic** [comment]
Thanks @elsaco, as a sanity check `wsl --version` are you on?

**Comment by pmartincic** [comment]
None of us have been able to repro this. If you have any clues that'd be appreciated.

**Comment by elsaco** [comment]
`FEFF` looks like a BOM signature. I can't tell where/how it was added to the `/etc/hosts` but it's gone now. Trying to setup `dnsmasq` would trigger `invalid entry in host file` error messages, otherwise I wouldn't notice it. Can't repro now.

**Comment by xanderificnl** [comment]
I'm seeing the same issue. You might be able to reproduce this issue by displaying non-printable characters, i.e. look for `M-oM-;M-?` by running `cat -A /etc/hosts`

Windows' host file starts with the offending character and it seems like it's being appended (albeit stripped from comments) to `/etc/hosts`

#### Get lines 6-9 from /etc/hosts and display non-printable characters
```bash
$ sed '6,9!d' /etc/hosts | cat -A
M-oM-;M-?$                                                                                      # <FEFF>
192.168.88.254^Ihost.docker.internal$
192.168.88.254^Igateway.docker.internal$
127.0.0.1^Ikubernetes.docker.internal$
```

#### Reformat Windows' host file
```bash
$ sed -f w32hosts.sed /mnt/c/Windows/System32/drivers/etc/hosts | cat -A
M-oM-;M-?$                                                                                      # <FEFF>
192.168.88.254^Ihost.docker.internal$
192.168.88.254^Igateway.docker.internal$
127.0.0.1^Ikubernetes.docker.internal$
```

##### w32hosts.sed

```sed
# Strip comments from
s/#.*//g
# Remove carriage returns
s/\r//g
# Remove empty lines
/^$/d
# Ensures spaces are tabulars
s/[[:space:]]/\t/g
```

**Comment by Tatsh** [comment]
Is Docker Desktop to blame for this? I do not think the file normally has a BOM at all. None of the other text files in the Windows etc directory have it.

**Comment by AnixPasBesoin** [comment]
This issue is still prevalent, any fix?

**Comment by OneBlue** [comment]
Fixed in [2.6.0](https://github.com/microsoft/WSL/releases/tag/2.6.0)
